File: frontend/src/app/model/users/supplier/supplier.ts
import {Gender} from "../patient/gender";
import {Address} from "../../address/address";
import {Role} from "../role";
import { Pharmacy } from "../../pharmacy/pharmacy";

export class Supplier{
  public name : String;
  public surname : String;
  public dateOfBirth : Date;
  public phoneNumber : String;
  public email : String;
  public password : String;
  public address : Address;
  public role : Role;
  public authorities : Number[];
  public firmName : String;
  public firstTimeChanged : boolean;

  constructor(name:String, surname : String, dateOfBirth: Date, phoneNumber : String, email : String, password : String,  address : Address, role : Role, autorities : Number[],  firmName : String,firstTimeChanged : boolean){
    this.name = name;
    this.surname = surname;
    this.dateOfBirth = dateOfBirth;
    this.phoneNumber = phoneNumber;
    this.email = email;
    this.password = password;
    this.address = address;
    this.role = role;
    this.authorities = autorities;
    this.firmName = firmName;
    this.firstTimeChanged = firstTimeChanged;
  }
}

File: src/main/java/com/atlaspharmacy/atlaspharmacy/supplier/domain/Supplier.java
package com.atlaspharmacy.atlaspharmacy.supplier.domain;

import com.atlaspharmacy.atlaspharmacy.generalities.domain.Address;
import com.atlaspharmacy.atlaspharmacy.users.domain.Authority;
import com.atlaspharmacy.atlaspharmacy.users.domain.SupplierAuthority;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;
import org.hibernate.annotations.Proxy;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "suppliers")
@Proxy(lazy = false)
public class Supplier {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    @OneToOne(fetch = FetchType.EAGER, cascade = CascadeType.MERGE)
    private Address headquarters;
    private String phoneNumber;
    private String email;
    private String password;
    @Column(insertable = false, updatable = false)
    private String role;
    @ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.MERGE)
    @JoinTable(name = "supplier_authority",
            joinColumns = @JoinColumn(name = "supplier_id", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name = "authority_id", referencedColumnName = "id"))
    @Fetch(value = FetchMode.SUBSELECT)
    private List<SupplierAuthority> authorities;

    public Supplier() {

    }

    public Supplier(Long id, String name, String phoneNumber, String email, String password, String role, List<SupplierAuthority> authorities) {
        this.id = id;
        this.name = name;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.password = password;
        this.role = role;
        this.authorities = authorities;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Address getAddress() {
        return headquarters;
    }

    public void setAddress(Address address) {
        this.headquarters = address;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public List<SupplierAuthority> getAuthorities() {
        return authorities;
    }

    public void setAuthorities(List<SupplierAuthority> authorities) {
        this.authorities = authorities;
    }
}
